webrtc/README.md

```diff
@@ -0,0 +1,490 @@
+overriding the default value of `ordered` `true` to `false` when creating a new
+data channel via
+[`RTCPeerConnection.createDataChannel`](https://www.w3.org/TR/webrtc/#dom-peerconnection-createdatachannel).
+
+### Head-of-line blocking
+
+WebRTC datachannels and the underlying SCTP is message-oriented and not
+stream-oriented (e.g. see
+[`RTCDataChannel.send()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/send)
+and
+[`RTCDataChannel.onmessage()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel#example)).
+libp2p streams on the other hand are byte oriented. Thus we run into the risk of
+head-of-line blocking.
+
+Given that the browser does not give us access to the MTU on a given connection,
+we can not make an informed decision on the optimal message size.
+
+We follow the recommendation of QUIC, requiring ["a minimum IP packet size of at
+least 1280
+bytes"](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-29#section-14).
+An SCTP packet common header is 12 bytes long. An SCTP data chunk header size is
+16 bytes. Thus implementations SHOULD choose a message size equal or below 1252
+bytes. Long term we hope to be able to give better recommendations based on
+real-world experiments.
+
+### `RTCDataChannel` negotiation
+
+`RTCDataChannel`s are negotiated in-band by the WebRTC user agent (e.g. Firefox,
+Pion, ...). In other words libp2p WebRTC implementations MUST NOT change the
+default value `negotiated: false` when creating a `RTCDataChannel` via
+`RTCPeerConnection.createDataChannel`.
+
+The WebRTC user agent (i.e. not the application) decides on the `RTCDataChannel`
+ID based on the local node's connection role. For the interested reader see
+[RF8832 Protocol
+Overview](https://www.rfc-editor.org/rfc/rfc8832.html#section-4). It is
+RECOMMENDED that user agents reuse IDs once their `RTCDataChannel` closes. IDs
+MAY be reused according to RFC 8831: "Streams are available for reuse after a
+reset has been performed", see [RFC 8831 6.7 Closing a Data Channel
+](https://datatracker.ietf.org/doc/html/rfc8831#section-6.7). Up to 65535
+(`2^16`) concurrent data channels can be opened at any given time.
+
+According to RFC 8832 a `RTCDataChannel` initiator "MAY start sending messages
+containing user data without waiting for the reception of the corresponding
+DATA_CHANNEL_ACK message", thus using `negotiated: false` does not imply an
+additional round trip for each new `RTCDataChannel`.
+
+### `RTCDataChannel` label
+
+`RTCPeerConnection.createDataChannel()` requires passing a `label` for the
+to-be-created `RTCDataChannel`. When calling `createDataChannel` implementations
+MUST pass an empty string. When receiving an `RTCDataChannel` via
+`RTCPeerConnection.ondatachannel` implementations MUST NOT require `label` to be
+an empty string. This allows future versions of this specification to make use
+of the `RTCDataChannel` `label` property.
```

Comment:
@ckousik as discussed out-of-band the `RTCDatachannel` `label` is now documented to be an empty string. Let me know in case I am missing something related to the Go and JS implementations.